### ISSUE TITLE
`invalid-return-type` for global variable

### ISSUE THREAD

**Issue by j178** [astral-sh/ty#416]
### Summary

It is common to use a function to return a global variable as a cache. Right now, `ty` is flagging errors on the type of the global variable. I also tried`mypy` and it passes.

```py
a = None

def f() -> int:
    global a
    if a is None:
        a = 1
    return a # error: Return type does not match returned value: expected `int`, found `Literal[1] | Unknown | None`
```

[playgroud](https://play.ty.dev/8e122ec4-47c0-474d-b529-c9d6a44063c0)

### Version

ty 0.0.1-alpha.3 (144a26d44 2025-05-15)
